bitsv/network/services/bitindex3.py

```diff
@@ -0,0 +1,150 @@
+        Gets utxo's for given legacy address
+        """
+        address = cashaddress.to_legacy_address(address)
+        r = requests.get(
+            f'https://api.bitindex.network/api/v3/{self.network}/addr/{address}/utxo',
+            headers=self.headers,
+        )
+        return [Unspent(
+            amount=tx['satoshis'],
+            confirmations=tx['confirmations'],
+            script=tx['scriptPubKey'],
+            txid=tx['txid'],
+            txindex=tx['vout'],
+        ) for tx in r.json()]
+
+    def get_balance(self, address):
+        """
+        Get address balances and transaction summary
+        """
+        address = cashaddress.to_legacy_address(address)
+        r = requests.get(
+            f'https://api.bitindex.network/api/v3/{self.network}/addr/{address}',
+            headers=self.headers,
+        )
+        return r.json()
+
+    def get_transactions(
+        self,
+        address,
+        fromIndex=0,
+        toIndex=0,
+        noAsm=True,
+        noScript=True,
+        noSpent=True,
+    ):
+        """
+        Retrieve transactions for an address
+        """
+        address = cashaddress.to_legacy_address(address)
+        json_payload = json.dumps({
+            "addrs": address,
+            "fromIndex": fromIndex,
+            "toIndex": toIndex,
+            "noAsm": noAsm,
+            "noScript": noScript,
+            "noSpent": noSpent,
+        })
+        r = requests.post(
+            f'https://api.bitindex.network/api/v3/{self.network}/addrs/txs',
+            data=json_payload,
+            headers=self.headers,
+        )
+        return r.json()
+
+    def send_transaction(self, rawtx):
+        """
+        Sends a transaction to the network
+        """
+        json_payload = json.dumps({"rawtx": rawtx})
+        r = requests.post(
+            f'https://api.bitindex.network/api/v3/{self.network}/tx/send',
+            data=json_payload,
+            headers=self.headers,
+        )
+        return r.json()
+
+    def get_transaction(self, txid):
+        """
+        Gets a single transaction
+        """
+        r = requests.get(
+            f'https://api.bitindex.network/api/v3/{self.network}/tx/{txid}',
+            headers=self.headers,
+        )
+        return r.json()['data']
+
+    def get_raw_transaction(self, txid):
+        """
+        Gets a single transaction as a raw string
+        """
+        r = requests.get(
+            f'https://api.bitindex.network/api/v3/{self.network}/rawtx/{txid}',
+            headers=self.headers,
+        )
+        return r.json()['rawtx']
+
+    def get_network_status(self, q='getInfo'):
+        """
+        Gets the network status
+
+        :param q: The type of status to query.
+        """
+        r = requests.get(
+            f'https://api.bitindex.network/api/v3/{self.network}/status?q={q}',
+            headers=self.headers,
+        )
+        return r.json()
+
+
+class BitIndex3MainNet(BitIndex3):
```

Comment:
These names are a bit long and clunky. I would love suggestions for better ones.

Also, not sure that this is even completely necessary. It would be easy to just use the `BitIndex3` class and just pass in a different network at initialization like `api = BitIndex3(api_key, 'test')`.

Reply:
I like it the way it is.

Reply:
ok!